src/libs/ReportUtils.ts

```diff
@@ -216,294 +488,218 @@ function isCanceledTaskReport(report = {}, parentReportAction = {}) {
 /**
  * Checks if a report is an open task report.
  *
- * @param {Object} report
- * @param {Object} parentReportAction - The parent report action of the report (Used to check if the task has been canceled)
- * @returns {Boolean}
+ * @param parentReportAction - The parent report action of the report (Used to check if the task has been canceled)
  */
-function isOpenTaskReport(report, parentReportAction = {}) {
-    return isTaskReport(report) && !isCanceledTaskReport(report, parentReportAction) && report.stateNum === CONST.REPORT.STATE_NUM.OPEN && report.statusNum === CONST.REPORT.STATUS.OPEN;
+function isOpenTaskReport(report: OnyxEntry<Report>, parentReportAction?: OnyxEntry<ReportAction> | EmptyObject): boolean {
+    return isTaskReport(report) && !isCanceledTaskReport(report, parentReportAction) && report?.stateNum === CONST.REPORT.STATE_NUM.OPEN && report?.statusNum === CONST.REPORT.STATUS.OPEN;
 }
 
 /**
  * Checks if a report is a completed task report.
- *
- * @param {Object} report
- * @returns {Boolean}
  */
-function isCompletedTaskReport(report) {
-    return isTaskReport(report) && report.stateNum === CONST.REPORT.STATE_NUM.SUBMITTED && report.statusNum === CONST.REPORT.STATUS.APPROVED;
+function isCompletedTaskReport(report: OnyxEntry<Report>): boolean {
+    return isTaskReport(report) && report?.stateNum === CONST.REPORT.STATE_NUM.SUBMITTED && report?.statusNum === CONST.REPORT.STATUS.APPROVED;
 }
 
 /**
  * Checks if the current user is the manager of the supplied report
- *
- * @param {Object} report
- * @returns {Boolean}
  */
-function isReportManager(report) {
-    return report && report.managerID === currentUserAccountID;
+function isReportManager(report: OnyxEntry<Report>): boolean {
+    return report?.managerID === currentUserAccountID;
 }
 
 /**
  * Checks if the supplied report has been approved
- *
- * @param {Object} report
- * @returns {Boolean}
  */
-function isReportApproved(report) {
-    return report && report.stateNum === CONST.REPORT.STATE_NUM.SUBMITTED && report.statusNum === CONST.REPORT.STATUS.APPROVED;
+function isReportApproved(report: OnyxEntry<Report>): boolean {
+    return report?.stateNum === CONST.REPORT.STATE_NUM.SUBMITTED && report.statusNum === CONST.REPORT.STATUS.APPROVED;
 }
 
 /**
  * Given a collection of reports returns them sorted by last read
- *
- * @param {Object} reports
- * @returns {Array}
  */
-function sortReportsByLastRead(reports) {
-    return _.chain(reports)
-        .toArray()
-        .filter((report) => report && report.reportID && report.lastReadTime)
-        .sortBy('lastReadTime')
-        .value();
+function sortReportsByLastRead(reports: OnyxCollection<Report>): Array<OnyxEntry<Report>> {
+    return Object.values(reports ?? {})
+        .filter((report) => report?.reportID && report?.lastReadTime)
```

Comment:
```suggestion
        .filter((report) => (!!report?.reportID) && (!!report?.lastReadTime))
```
Maybe something on the lines of this here